packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.spec.tsx

```diff
@@ -264,6 +264,53 @@ describe('<ArrayInput />', () => {
         expect(screen.queryByText('test helper text')).not.toBeNull();
     });
 
+    it('should update the form state to dirty, and allow submit, on updating an array input with default value', async () => {
```

Comment:
Although the test is not as clean as I would like it does protect against the bug identified successfully.